cdap-ui/app/features/tracker/controllers/integrations-ctrl.js

```diff
@@ -0,0 +1,244 @@
+      namespace: this.$state.params.namespace,
+      scope: this.$scope
+    };
+    this.myTrackerApi.getNavigatorApp(params)
+      .$promise
+      .then((res) => {
+        this.navigatorSetup.isSetup = true;
+        let config = {};
+        try {
+          config = JSON.parse(res.configuration);
+
+          this.navigatorInfo.hostname = config.navigatorConfig.navigatorHostName;
+          this.navigatorInfo.username = config.navigatorConfig.username;
+          this.navigatorInfo.password = config.navigatorConfig.password;
+          this.navigatorInfo.brokerString = config.metadataKafkaConfig.brokerString;
+
+          this.getNavigatorStatus();
+        } catch (e) {
+          console.log('Cannot parse configuration JSON');
+        }
+      }, () => {
+        this.getKafkaBrokerList();
+      });
+  }
+
+  getNavigatorStatus() {
+    let params = {
+      namespace: this.$state.params.namespace,
+      scope: this.$scope
+    };
+
+    this.myTrackerApi.getNavigatorStatus(params)
+      .$promise
+      .then((res) => {
+        if (res.length) {
+          this.navigatorState = res[0];
+
+          if (res[0].status === 'RUNNING') {
+            this.navigatorSetup.isEnabled = true;
+            this.fetchMetrics();
+          } else {
+            this.dataSrc.stopPoll(this.pollId.__pollId__);
+          }
+        } else {
+          this.navigatorState = {
+            status: 'DISABLED'
+          };
+        }
+      });
+  }
+
+  toggleNavigator() {
+    let params = {
+      namespace: this.$state.params.namespace,
+      action: this.navigatorSetup.isEnabled ? 'start' : 'stop',
+      scope: this.$scope
+    };
+
+    this.myTrackerApi.toggleNavigator(params, {})
+      .$promise
+      .then( () => {
+        this.getNavigatorStatus();
+      }, (err) => {
+        this.myAlertOnValium.show({
+          type: 'danger',
+          content: err.data
+        });
+      });
+  }
+
+  fetchMetrics() {
+
+    let metric = {
+      startTime: 'now-1h',
+      endTime: 'now',
+      resolution: '1m',
+      names: ['system.process.events.processed']
+    };
+
+    let tags = {
+      namespace: this.$state.params.namespace,
+      app: 'ClouderaNavigator',
+      flow: 'MetadataFlow'
+    };
+
+    // fetch timeseries metrics
+    this.pollId = this.dataSrc.poll({
+      _cdapPath: '/metrics/query',
+      method: 'POST',
+      body: this.MyMetricsQueryHelper.constructQuery('qid', tags, metric)
+    }, (res) => {
+      let processedData = this.MyChartHelpers.processData(
+        res,
+        'qid',
+        metric.names,
+        metric.resolution
+      );
+
+      processedData = this.MyChartHelpers.c3ifyData(processedData, metric, metric.names);
+      this.chartData = {
+        x: 'x',
+        columns: processedData.columns,
+        keys: {
+          x: 'x'
+        }
+      };
+
+      this.eventsSentAggregate = _.sum(this.chartData.columns[0]);
+    });
+  }
+
+  saveNavigatorSetup() {
+    let params = {
```

Comment:
May be we should do a check here or make the save button disabled when no fields are entered